source/speech/speech.py

```diff
@@ -1169,9 +1170,28 @@ def speak(  # noqa: C901
 			)
 			if not inCharacterMode:
 				speechSequence[index] += CHUNK_SEPARATOR
+				if curLanguage not in (defaultLanguage, SpeechState.lastReportedLanguage):
+					speechSequence.insert(0, LangChangeCommand(defaultLanguage))
+					speechSequence.insert(1, languageHandler.getLanguageDescription(curLanguage))
+					SpeechState.lastReportedLanguage = curLanguage
+					if not languageIsSupported(curLanguage):
+						log.warn(f"{curLanguage} not supported in {getSynth().name}")
```

Comment:
Seems that `log.warn` is usually not used in NVDA's code base. Any specific reason to use it?

We usually have `log.warning` (used even at info level), and `log .debugWarning`, used at the same log level.

Reply:
Just a mistake from my side. My intention was to use log.debugWarning